nexus/db-queries/src/db/datastore/migration.rs

```diff
@@ -38,6 +42,22 @@ impl DataStore {
             .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))
     }
 
+    /// List all migrations associated with the provided instance ID.
+    pub async fn migration_list_by_instance(
```

Comment:
Just curious, is there ever more than one entry here? We're filtering to non-deleted migrations, so I naively expected to see just a single value returned here.

Reply:
There's never more than one entry *yet*.  Right now, migrations are marked as deleted as soon as the migration completes. However, this will probably change in the future, and migrations will be deleted only when the instance itself is deleted (and possibly once they have been completed for a 'fairly long' period of time). I didn't make that part of the change in this PR to keep the diff small, but I suppose I could...

Reply:
Interesting, thanks for the explanation! Is the idea to keep them around in the future as an auditable log of migrations? I have no opinion on whether that should be added now, but I do think a quick comment that there may be more than one in the future might be helpful. Thanks!

Reply:
That's part of the motivation, yeah. Additionally, though, migration records will be used by the `instance-update` saga to do things like determining which Propolis processes are no longer in use because their instance has been migrated out.